### ISSUE TITLE
Number (and currency?) component 'allowNegative' option does not make sense

### ISSUE THREAD

**Issue by sergei-maertens** [open-formulieren/open-forms#3430]
### Product versie / Product version

all

### Omschrijf het probleem / Describe the bug

Specifically looking at the number component here, but this (likely) applies to Currency too (and is probably the reason this exists).

Number components allow setting three validators:

- required
- min
- max

On top of that, there are "display" options: decimal limit (0, 1, 2, 3...) and allow negative (boolean). The latter does not make sense when combined with the min/max validator: you could configure the component to not allow negative values, but also specify a maximum value of -5 -> this gives an empty possible value set.

We do suspect this is related to the currency component not supporting negative values out of the box and our hack/workaround to allow this in a backwards compatible manner, since the number formatting of Form.io is very weird and frustrating (see also #2341 ). This behaviour is also not accessible - user input is silently ignored (the minus sign), or decimals from numbers are truncated without any feedback. It would probably make more sense if those options result in validation errors that can be accessed by screen readers.

Proposed changes:

* remove `allowNegative`, instead if it's not allowed -> set the `validate.min` to the value `0` so that you only have positive (and zero) values. If it's checked, do nothing - unspecified min/max covers this situation.
* move `decimalLimit` to the validation tab and set up a proper validator for it (in the event that the limit is 0). We need to decide what to do with non-zero limits where the user *does* provide more precise input - possibly some aria-polite warnings can help?

### Screen resolution

None

### Device

None

### OS

None
